rasa/core/test.py

```diff
@@ -649,13 +649,13 @@ def _collect_action_executed_predictions(
     policy_entity_result = None
     prev_action_unlikely_intent = False
 
-    if circuit_breaker_tripped:
-        prediction = PolicyPrediction([], policy_name=None)
-        predicted_action = "circuit breaker tripped"
-    else:
+    try:
         predicted_action, prediction, policy_entity_result = _run_action_prediction(
             processor, partial_tracker, expected_action
         )
+    except ActionLimitReached:
+        prediction = PolicyPrediction([], policy_name=None)
+        predicted_action = "circuit breaker tripped"
```

Comment:
Yes, I think it makes to apply this try-except to the case when `action_unlikely_intent` is predicted and another prediction needs to be made (line 674)